packages/engine-cli/src/top-level-config-plugin-esbuild.ts

```diff
@@ -86,5 +86,6 @@ function emitConfigFile(
     const content = JSON.stringify(imported.default ?? imported);
     const configFileName = envName ? `${fileName!}.${envName}` : fileName;
     const configPath = `configs/${configFileName!}.json`;
-    filesToEmit.set(configPath, { content, path: join(rootContext, outDir, configPath) });
+    const path = isAbsolute(outDir) ? join(outDir, configPath) : join(rootContext, outDir, configPath);
```

Comment:
use path.resolve in such cases, instead of checking for absolute yourself:
`path.resolve(rootContext, outDir, configPath)`